### ISSUE TITLE
Debugger support for discards

### ISSUE THREAD

**Issue by miloush** [dotnet/roslyn#18202]
First I want to commend the decision that the `_` is not a discard when declared as a variable. I haven't been using `_` as standalone variables before, but I started using this "variable feature" for the purpose of debugging discards. It allows me to hover over the `_` during debugging and see the "discarded" value, without any modifications to that code.

I wonder whether if people would find useful to have the discarded values in the Auto window just like we have it now for (also often discarded) return values of the methods, or perhaps even the debug tips on hover, without the need to declare the variable?

**Comment by jnm2** [comment]
Could you work around collisions?

```c#
_ = Foo(); // I think debugger expressions are evaluated as text,
_ = Bar(); // so how does it tell apart "_" and "_" when you hover?
```

**Comment by miloush** [comment]
Well, one option is to remember the last one, effectively the same behavior as if `object _` had been declared above your lines.

The watch/auto window could probably be supplied with more information from the call site, but I agree that if the debugger only uses the symbol name and that gets to be "_" in this case, the hovering part might not be feasible otherwise.

**Comment by miloush** [comment]
Also the debug tip for `var` displays the actual type on hover, so I guess there si more than just a string being passed around?

**Comment by jnm2** [comment]
@miloush I think that it also queries the type based on the string `"_"`. You can see this in action:

```c#
// This is a comment, but select this character: _
// Then right click and click Pin or add Watch
// The type comes through even though it originated in a comment.
var _ = Foo();
```

You can see the same thing if you move to a different stack frame:
```c#
void RunFoo()
{
    var _ = Foo();
    RunBar();
}

void RunBar()
{
    var _ = Bar();
    Debugger.Break(); // Hovering over both "_" variables shows Bar objects,
    // unless you select a different stack frame. Then hovering over both variablse shows Foo objects.
}
```

**Comment by miloush** [comment]
@jnm2 yes, I agree, but that wasn't my point. My point was that if you do
```C#
var a = 4;
var b = "";
```
then you can hover over `var` and they will show different types (even in the same stack frame). To be fair though, that's probably a different mechanism than debug time tips. Either way, discussing this sounds more like an implementation detail of the proposed feature.

**Comment by jnm2** [comment]
Ah, gotcha. Never noticed that before.

**Comment by jrmoreno1** [comment]
Just came here to make this suggestion, but found this when checking to see if it had already been made.  Either that or a way to make the pragma to disable the warning for an unused variable only apply to the next line.  

I need the value when debugging, the program doesn't need it to execute.
